runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -1480,6 +1445,9 @@ TR::Register *J9::X86::TreeEvaluator::multianewArrayEvaluator(TR::Node *node, TR
 
    deps->stopAddingConditions();
 
+   generateLabelInstruction(LABEL, node, oolJumpPoint, cg);
```

Comment:
Am I reading the code wrong, or does the loop exit on L1413 (`generateLabelInstruction(JA4, node, loopLabel, cg);`) fall into this code if the backedge isn't taken, which will always take you to the OOL instructions?  Is this the behaviour you expect because I don't think that's how it originally behaved.  You might need a branch around this if not.

Reply:
Good catch! We'll need a jump to the `fallThru` label on line 1451 if we don't branch back to `loopLabel`. I've added it here: dde60a6408

Reply:
Let's run the microbenchmark on x86 to ensure we don't regress performance.

Reply:
Here are the results from the microbenchmark (performance looks as expected):
```
firstDimLength=0, secondDimLength=0,maxIterations=1000000 (allocation done in jit code)

   -- finale --
   42.2s:  >! 155.3M    167.6M    9.0   155.3M  155.3M  1886.1
   43.2s:  <  175.2M    166.6M    5.4   155.3M  155.3M  1886.1
   44.2s:  >  162.1M    163.7M    3.2   162.1M  162.1M  1890.4
   45.2s:  >! 166.3M    169.3M    3.4   166.3M  166.3M  1892.9
   46.2s:  <  172.1M    170.5M    2.0   166.3M  166.3M  1892.9
   47.2s:  >  168.8M    169.6M    1.2   168.8M  168.8M  1894.4
   48.2s:  <  170.7M    164.9M    2.8   168.8M  168.8M  1894.4
   49.2s:  >  162.6M    167.0M    1.7   168.8M  168.8M  1894.4
   50.2s:  <  168.4M    167.8M    1.0   168.8M  -inf    --
   51.2s:  >  166.9M    168.5M    0.6   168.8M  166.9M  1893.3
   52.2s:  <  169.0M    167.2M    0.8   168.8M  166.9M  1893.3
   53.2s:  >  166.5M    169.5M    1.4   168.8M  166.9M  1893.3  
MultiANewArray score: 168802736.000000 (168.8M 1894.4%)
           uncertainty:   1.4%
```

```
firstDimLength=1000,secondDimLength=0,maxIterations=1000000 (allocation done in JIT code)

   -- finale --
   36.2s:  <  33.93K    33.85K    1.3   33.49K  33.49K  1041.9
   37.2s:  >  33.64K    33.92K    0.8   33.64K  33.64K  1042.4
   38.2s:  <  34.05K    33.72K    0.5   33.64K  33.64K  1042.4
   39.3s:  <! 33.65K    30.13K   10.7   33.64K  33.64K  1042.4
   40.2s:  >  28.52K    33.62K   12.2   33.64K  33.64K  1042.4
   41.2s:  <  35.68K    33.51K    7.3   33.64K  33.64K  1042.4
   42.2s:  >  32.28K    32.76K    4.4   33.64K  33.64K  1042.4
   43.2s:  <  33.48K    33.31K    2.6   33.64K  -inf    --
   44.2s:  >  32.87K    33.22K    1.6   33.64K  32.87K  1040.0
   45.2s:  >! 33.48K    33.64K    1.9   33.64K  33.48K  1041.9
   46.2s:  <  33.96K    33.67K    1.1   33.64K  33.48K  1041.9
   47.2s:  >  33.47K    33.61K    0.7   33.64K  33.48K  1041.9  
MultiANewArray score: 33642.914063 (33.64K 1042.4%)
           uncertainty:   0.7%
```

```
firstDimLength=1000,secondDimLength=1,maxIterations=1000000 (jump to OOL to do allocation)

 -- finale --
   38.8s:  <  6870      6685      1.6   6585    6585    879.3
   39.8s:  <! 6632      6566      1.8   6585    6585    879.3
   40.8s:  >  6507      6631      1.1   6585    6585    879.3
   41.7s:  >! 6667      6695      1.3   6667    6667    880.5
   42.8s:  <  6738      6589      1.6   6667    6667    880.5
   43.7s:  >  6538      6823      3.6   6667    6667    880.5
   44.8s:  <  6944      6732      2.1   6667    6667    880.5
   45.7s:  >  6660      6776      1.3   6667    6667    880.5
   46.7s:  >! 6820      6825      1.5   6820    6820    882.8
   47.9s:  <  6877      6181      9.4   6820    6820    882.8
   48.7s:  >  5891      6764      9.9   6820    6820    882.8
   49.8s:  <  7098      6768      5.9   6820    6820    882.8
   50.8s:  >  6567      6701      3.6   6820    6820    882.8
  MultiANewArray score: 6819.545410 (6820 882.8%)
           uncertainty:   3.6%
```